core/controllers/reader.py

```diff
@@ -288,7 +288,8 @@ def get(self, exploration_id):
                 rights_manager.ACTIVITY_TYPE_EXPLORATION, exploration_id):
             raise self.PageNotFoundException
 
-        is_iframed = (self.request.get('iframed') == 'true')
+        url_parts = [x.strip() for x in self.request.path.split('/')]
```

Comment:
Instead of doing this, can main.py link to a completely separate controller in this file for the /embed URL? So at this point there will be two separate controllers.

Then, find a way to factor the common logic out of the controllers into a common top-level function. So, for example, each controller may look something like this:

    _check_exploration_validity(exploration_id, version, collection_id)
    dependencies_html, gadget_templates, interaction_templates = _get_exploration_components(exploration_id)
    self.values.update(...)
    self.render_template(...)

and most of the heavy logic goes into top-level functions that are used for both controllers. This is what I meant by the refactoring.